### PR TITLE
Make feature specs less flappy

### PR DESCRIPTION
**Pull request by mjgiarlo**
# Why was this change made? 🤔

Less time re-running builds == more efficiency & developer happiness

# How was this change tested? 🤨

CI, local

# Does your change introduce accessibility violations? 🩺

No

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -109,6 +109,7 @@ end
 
 group :test do
   gem "capybara"
+  gem "capybara-screenshot"
```
Comment (justinlittman) [inline Gemfile:112]: I think `capybara-screenshot` is included transitively.

Comment (mjgiarlo) [reply Gemfile:112]: I don't believe it was in the `Gemfile.lock` prior to the changes in this branch. I can try commenting this out, re-bundling, and seeing whether the `require` still works though?

Comment (mjgiarlo) [reply Gemfile:112]: The `require` fails without `capybara-screenshot` in the Gemfile, alas.

File: .circleci/config.yml [hunk 2]
```diff
@@ -241,3 +241,7 @@ jobs:
       - run:
           name: Report test coverage results to CodeClimate
           command: ./cc-test-reporter after-build --coverage-input-type simplecov --exit-code $?
+      - store_artifacts:
+          path: tmp/capybara
+      - store_artifacts:
+          path: log/test.log
```
Comment (mjgiarlo) [inline .circleci/config.yml:247]: Stuff capybara screenshots and the Rails log in CI so we can take a look when tests fail.

File: Gemfile (hunk 1 above)
Comment (mjgiarlo) [inline Gemfile:112]: Help diagnose failures in the future.

File: spec/features/date_range_form_spec.rb [hunk 3]
```diff
@@ -39,6 +39,7 @@
 
     before do
       visit root_path(query_params.merge(all: true))
+      scroll_to find_by_id("facet-object_modified_date-header")
```
Comment (mjgiarlo) [inline spec/features/date_range_form_spec.rb:42]: Help the browser find the button on the next line.

File: spec/features/date_range_form_spec.rb [hunk 4]
```diff
@@ -100,6 +101,7 @@
...
```
Comment (mjgiarlo) [inline spec/features/date_range_form_spec.rb:104]: Help the browser find the button on the next line.

File: spec/features/item_registration_spec.rb [hunk 5]
```diff
@@ -125,7 +125,12 @@
 
       expect(page).to have_content "Items successfully registered."
 
-      find("td > a").click # Click on the druid link in the table
+      druid_link = find("td > a")
+
+      # NOTE: Unless we force a reindex here, this spec is flappy (example RSpec seed: 39128)
```
Comment (mjgiarlo) [inline spec/features/item_registration_spec.rb:130]: See comment.

File: spec/features/item_view_spec.rb [hunk 6]
```diff
@@ -270,6 +270,8 @@
 
             # The following three clicks make sure events are expandable and collapsible
             click_button "Events"
+            # NOTE: Without scrolling to the events section, the clicks below were flappy. (example RSpec seed: 39192)
```
Comment (mjgiarlo) [inline spec/features/item_view_spec.rb:273]: See comment.

File: spec/support/selenium_webdriver.rb [hunk 7]
```diff
@@ -2,20 +2,21 @@
...
 end
 
-Capybara.disable_animation = true
 Capybara.default_max_wait_time = 7
+Capybara.disable_animation = true
+Capybara.javascript_driver = :headless_chrome
+
+Capybara::Screenshot.register_driver(:headless_chrome) do |driver, path|
+  driver.browser.save_screenshot(path)
+end
```
Comment (mjgiarlo) [inline spec/support/selenium_webdriver.rb:22]: This is mostly:

1. Setting up capybara-screenshot
2. Updating the syntax
3. Removing 1-2 lines of cruft
4. Re-ordering the code a bit
5. Specifying a slightly larger size to make sure feature tests are getting the full-width website and not a responsive version (e.g., with an unexpected hamburger menu)

**Review by justinlittman — APPROVED** [review]

**Pull request merged**